portfolio/src/main/java/com/google/sps/servlets/PokedexDataServlet.java

```diff
@@ -0,0 +1,49 @@
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.Scanner;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/pokedex-data")
+public class PokedexDataServlet extends HttpServlet {
+  private static final int NUMBER_OF_GENERATIONS =
+      7; // 6 generations, Array Index directly  (1st Gen -> index of 1)
+
+  private int getColIndex(String header, String colName) {
+    String[] colNames = header.split(",");
+    for (int i = 0; i < colNames.length; i++) {
+      if (colName.equals(colNames[i])) {
+        return i;
+      }
+    }
+    System.error.println("ERROR: Column Name given not valid")
```

Comment:
I would add the "colName" and "header" params to this error message, since it would make debugging a lot easier if you ever see this error message logged. E.g. something like:

System.error.println("ERROR: Could not find colName " + colName + "in header " + header)

You could also try using the String.format() method if you're up for it:
https://docs.oracle.com/javase/7/docs/api/java/lang/String.html#format(java.lang.String,%20java.lang.Object...)